Cython/Compiler/PyrexTypes.py

```diff
@@ -5161,6 +5161,36 @@ def real_type(ntype):
     return widest_type
 
 
+
+def result_type_of_builtin_operation(builtin_type, type2):
+    """
+    Try to find a suitable (C) result type for a binary operation with a known builtin type.
+    """
+    if builtin_type.name == 'float':
+        if type2.is_numeric:
+            return widest_numeric_type(c_double_type, type2)
+        elif type2.is_builtin_type and type2.name == 'int':
```

Comment:
This feels like it goes against the general rule of writing things like `if type2 is Builtin.int_type`. I assume this is just to avoid circular dependencies?